src/GameMachineController.java

```diff
@@ -0,0 +1,28 @@
+import java.util.Optional;
+import java.util.Scanner;
+
+public class GameMachineController {
+
+    private Scanner sc = new Scanner(System.in);
+    private GameMachine machine = new GameMachine();
+
+    public void GameShop() {
+
+        System.out.println("Give game's name which you want buy:");
+        String wantedGame = sc.nextLine();
+        System.out.println("Pay:");
+        double payment = sc.nextDouble();
+        sc.nextLine();
+
+        try {
+            Game game = machine.getGame(wantedGame);
```

Comment:
I would add a new method to machine. buyGame which would get a game and pay for it in one go. It makes much more sense than invoking getGame and then paidForGame. Your solution seems a little bit not natural

Reply:
I add new method and remove old ones as you said.